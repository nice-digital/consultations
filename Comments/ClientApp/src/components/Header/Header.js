// @flow

import React, { Fragment, PureComponent } from "react";
import Moment from "react-moment";
import { Link } from "react-router-dom";
import { Alert } from "@nice-digital/nds-alert";

type PropsType = {
	title: string,
	endDate?: any,
	match?: any,
	subtitle1?: string,
	subtitle2?: string,
	consultationState?: {
		endDate: string,
		consultationIsOpen: boolean,
		consultationHasNotStartedYet?: boolean,
	},
}

export class Header extends PureComponent<PropsType> {

	render() {
		const {
			title,
			subtitle1,
			subtitle2,
		} = this.props;

		let startDate, endDate, isOpen, notStartedYet;

		if (this.props.consultationState) {
			startDate = this.props.consultationState.startDate || "";
			endDate = this.props.consultationState.endDate || "";
			isOpen = this.props.consultationState.consultationIsOpen || "";
			notStartedYet = this.props.consultationState.consultationHasNotStartedYet || "";
		}

		let startOrEnd = "ended";
		let showStartDate = false;

		if (notStartedYet) {
			startOrEnd = "starts";
			showStartDate = true;
		}

		return (
			<Fragment>
				<h1 data-qa-sel="changeable-page-header" className="page-header__heading mt--0">{title}</h1>
				{this.props.consultationState &&
					<div className="mb--d">
						<p className="container container-full ml--0">
							{isOpen ?
								<span>
									<span className="tag tag--open">Open for comments</span> Open until{" "}
									<Moment format="D MMMM YYYY" date={endDate}/>
								</span>
								:
								<span>
									{startOrEnd === "starts" && <span className="tag">Not yet open for comments</span>}
									{startOrEnd === "ended" && <span className="tag">Closed for comments</span>}{" "}
									This consultation {startOrEnd} on <Moment format="D MMMM YYYY" date={(showStartDate ? startDate : endDate)}/> at{" "}
									<Moment format="HH:mm" date={(showStartDate ? startDate : endDate)}/>
								</span>
							}
							&nbsp;&nbsp;
							<Link to={"/leadinformation"}>
								Request commenting lead permission
							</Link>
						</p>
					</div>
				}
				{subtitle1 && <p>{subtitle1}</p>}
				{subtitle2 &&
					<Alert type="caution" role="alert">
						<p>{subtitle2}</p>
					</Alert>
				}
			</ Fragment>
		);
	}

}
