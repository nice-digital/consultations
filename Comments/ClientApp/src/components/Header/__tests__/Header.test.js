import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { Header } from "../Header";

test("should display correctly when consulation hasn't started", () => {
	const dateNow = Date.now;
	Date.now = jest.fn(() => new Date(Date.UTC(2022, 7, 9, 8)).valueOf());
	const FakeProps = {
		title: "My title",
		subtitle1: "Subtitle 1",
		subtitle2: "Subtitle 2",
		consultationState: {
			endDate: "1994-11-05T08:15:30-05:00",
			consultationIsOpen: false,
			consultationHasNotStartedYet: true,
		},
	};
	const {container} = render(<MemoryRouter><Header {...FakeProps}/></MemoryRouter>);
	expect(container).toMatchSnapshot();
	Date.now = dateNow;
});

test("should display correctly when consulation is in progress", () => {
	const FakeProps = {
		title: "My title",
		subtitle1: "Subtitle 1",
		consultationState: {
			endDate: "1994-11-05T08:15:30-05:00",
			consultationIsOpen: true,
			consultationHasNotStartedYet: false,
		},
	};
	const {container} = render(<MemoryRouter><Header {...FakeProps}/></MemoryRouter>);
	expect(container).toMatchSnapshot();
});

test("should display correctly when consulation has ended", () => {
	const FakeProps = {
		title: "My title has changed",
		subtitle1: "Subtitle 1",
		consultationState: {
			endDate: "1994-11-05T08:15:30-05:00",
			consultationIsOpen: false,
			consultationHasNotStartedYet: false,
		},
	};
	const {container} = render(<MemoryRouter><Header {...FakeProps}/></MemoryRouter>);
	expect(container).toMatchSnapshot();
});
