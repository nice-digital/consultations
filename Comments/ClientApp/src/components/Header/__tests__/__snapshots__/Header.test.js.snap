// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ClientApp]  Header  should display correctly when consulation has ended 1`] = `
<Fragment>
  <h1
    className="page-header__heading mt--0"
    data-qa-sel="changeable-page-header"
  >
    My title has changed
  </h1>
  <div
    className="mb--d"
  >
    <p
      className="container container-full ml--0"
    >
      <span>
        <span
          className="tag"
        >
          Closed for comments
        </span>
         
        This consultation 
        ended
         on 
        <t
          ago={false}
          calendar={false}
          date="1994-11-05T08:15:30-05:00"
          decimal={false}
          element="time"
          filter={[Function]}
          format="D MMMM YYYY"
          fromNow={false}
          interval={60000}
          onChange={[Function]}
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
        />
         at
         
        <t
          ago={false}
          calendar={false}
          date="1994-11-05T08:15:30-05:00"
          decimal={false}
          element="time"
          filter={[Function]}
          format="HH:mm"
          fromNow={false}
          interval={60000}
          onChange={[Function]}
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
        />
      </span>
        
      <Link
        replace={false}
        to="/leadinformation"
      >
        Request commenting lead permission
      </Link>
    </p>
  </div>
  <p>
    Subtitle 1
  </p>
</Fragment>
`;

exports[`[ClientApp]  Header  should display correctly when consulation hasn't started 1`] = `
<Fragment>
  <h1
    className="page-header__heading mt--0"
    data-qa-sel="changeable-page-header"
  >
    My title
  </h1>
  <div
    className="mb--d"
  >
    <p
      className="container container-full ml--0"
    >
      <span>
        <span
          className="tag"
        >
          Not yet open for comments
        </span>
         
        This consultation 
        starts
         on 
        <t
          ago={false}
          calendar={false}
          date=""
          decimal={false}
          element="time"
          filter={[Function]}
          format="D MMMM YYYY"
          fromNow={false}
          interval={60000}
          onChange={[Function]}
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
        />
         at
         
        <t
          ago={false}
          calendar={false}
          date=""
          decimal={false}
          element="time"
          filter={[Function]}
          format="HH:mm"
          fromNow={false}
          interval={60000}
          onChange={[Function]}
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
        />
      </span>
        
      <Link
        replace={false}
        to="/leadinformation"
      >
        Request commenting lead permission
      </Link>
    </p>
  </div>
  <p>
    Subtitle 1
  </p>
  <Alert
    role="alert"
    type="caution"
  >
    <p>
      Subtitle 2
    </p>
  </Alert>
</Fragment>
`;

exports[`[ClientApp]  Header  should display correctly when consulation is in progress 1`] = `
<Fragment>
  <h1
    className="page-header__heading mt--0"
    data-qa-sel="changeable-page-header"
  >
    My title
  </h1>
  <div
    className="mb--d"
  >
    <p
      className="container container-full ml--0"
    >
      <span>
        <span
          className="tag tag--open"
        >
          Open for comments
        </span>
         Open until
         
        <t
          ago={false}
          calendar={false}
          date="1994-11-05T08:15:30-05:00"
          decimal={false}
          element="time"
          filter={[Function]}
          format="D MMMM YYYY"
          fromNow={false}
          interval={60000}
          onChange={[Function]}
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
        />
      </span>
        
      <Link
        replace={false}
        to="/leadinformation"
      >
        Request commenting lead permission
      </Link>
    </p>
  </div>
  <p>
    Subtitle 1
  </p>
</Fragment>
`;
